knowledge-base/grid-get-index-of-grid-row.md

```diff
@@ -0,0 +1,202 @@
+    </GridColumns>
+</TelerikGrid>
+
+@if (!String.IsNullOrEmpty(Result))
+{
+    <div>
+        @Result
+    </div>
+}
+
+@code {
+    public int Total { get; set; }
+    List<MainModel> salesTeamMembers { get; set; }
+    IEnumerable<MainModel> selectedItems { get; set; } = new List<MainModel>();
+    List<MainModel> currentSalesTeamMembers { get; set; } //implementation of OnRead
+    MainModel currentItem { get; set; } //need to save the selected item to determine its new index
+    public string Result { get; set; }
+
+    void SelectedItemsChanged(IEnumerable<MainModel> items)
+    {
+        selectedItems = items;
+
+        var selectedItem = items.FirstOrDefault();
+
+        currentItem = selectedItem;
+    }
+
+    async Task OnReadHandler(GridReadEventArgs args)
+    {
+        var dataSourceResult = salesTeamMembers.ToDataSourceResult(args.Request);
+
+        currentSalesTeamMembers = dataSourceResult.Data.Cast<MainModel>().ToList(); // this is the collection with the sorted items
+
+        int currentIndexOfItem = currentSalesTeamMembers.IndexOf(currentItem);
+
+        if(currentIndexOfItem > -1) // this check is not mandatory
+        {
+            Result = $"The index of the selected item is: {currentIndexOfItem}";
+        }
+
+        Total = dataSourceResult.Total;
+
+        StateHasChanged();
+    }
+
+    protected override void OnInitialized()
+    {
+        salesTeamMembers = GenerateData();
+    }
+
+    private List<MainModel> GenerateData()
+    {
+        List<MainModel> data = new List<MainModel>();
+        for (int i = 0; i < 5; i++)
+        {
+            MainModel mdl = new MainModel { Id = i, Name = $"Name {i}" };
+            data.Add(mdl);
+        }
+        return data;
+    }
+
+    public class MainModel
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+    }
+}
+````
+
+### Get the New Index Of a Clicked Row
+
+````CSHTML
+@*Get the new index of a clicked row*@
+
+@using Telerik.DataSource
+@using Telerik.DataSource.Extensions
+
+<TelerikGrid Data="@currentSalesTeamMembers"
+             OnRowClick="@OnRowClickHandler"
+             OnRead="@OnReadHandler"
+             TotalCount="@Total"
+             Sortable="true">
+    <GridColumns>
+        <GridColumn Field="Id"></GridColumn>
+        <GridColumn Field="Name"></GridColumn>
+    </GridColumns>
+</TelerikGrid>
+
+@if (!String.IsNullOrEmpty(Result))
+{
+    <div>
+        @Result
+    </div>
+}
+
+@code {
+    public int Total { get; set; }
+    List<MainModel> salesTeamMembers { get; set; }
+    List<MainModel> currentSalesTeamMembers { get; set; } //implementation of OnRead
+    MainModel currentItem { get; set; } //need to save the selected item to determine its new index
+    public string Result { get; set; }
+
+    void OnRowClickHandler(GridRowClickEventArgs args)
+    {
+        var clickedRow = args.Item as MainModel;
+
+        currentItem = clickedRow;
```

Comment:
add a method call here that will get the index immediately - you already have the current data because of onread, so you should move the logic here